WeatherPlannerDemo/App/Data/CoreData/CoreDataService.swift

```diff
@@ -10,60 +10,53 @@ class CoreDataService: CoreDataServiceProtocol {
     
     // MARK: - Fetches
     
-    func fetchCities() -> [CDCity] {
+    func fetchCities() -> Observable<[CDCity]> {
         let request: NSFetchRequest<CDCity> = CDCity.fetchRequest()
+        request.sortDescriptors = [NSSortDescriptor(key: "name", ascending: true)]
         request.returnsObjectsAsFaults = false
         
-        do {
-            return try coreDataStack.mainContext.fetch(request)
-        } catch let error {
-            Logger.print(string: "Failed fetching data from database: \(error.localizedDescription)")
-            return []
-        }
+        return coreDataStack.mainContext.rx_entities(request as? NSFetchRequest<NSFetchRequestResult>)
+            .flatMap { fetchedManagedObject -> Observable<[CDCity]> in
+                guard let fetchedCities = fetchedManagedObject as? [CDCity] else { return .just([]) }
+                return .just(fetchedCities)
+            }
     }
     
-    func fetchCityWithID(_ id: Int) -> CDCity {
-        let request: NSFetchRequest<CDCity> = CDCity.fetchRequest()
-        request.predicate = Predicates.idPredicate(id)
-        request.returnsObjectsAsFaults = false
+    func fetchAirPollutionForCity(withID id: Int) -> Observable<CDAirPollution?> {
+        
+        guard let city = self.fetchCityWithID(id) else { return .just(nil) }
         
-        do {
-            return try coreDataStack.mainContext.fetch(request).first ?? CDCity()
-        } catch let error {
-            Logger.print(string: "Failed fetching data from database: \(error.localizedDescription)")
-            return CDCity()
-        }
-    }
-    
-    func fetchAirPollutionForCity(_ city: CDCity) -> CDAirPollution? {
         let request: NSFetchRequest<CDAirPollution> = CDAirPollution.fetchRequest()
         request.predicate = Predicates.airPollutionPredicate(city)
+        request.sortDescriptors = [NSSortDescriptor(key: "aqi", ascending: true)]
```

Comment:
What is aqi?

Reply:
air quality index.. each city has 1 CDAirPollution, and each CDAirPollution has aqi attribute... since it has only 1 air quality index, basically I do not need that sortDescriptor, but if it's not specified I get error - each fetch request must have sort descriptor